meson.build

```
@@ -0,0 +1,266 @@
+  'util.c',
+  'bitvector.c',
+  'mdebug.c'])
+
+incl_dirs = include_directories(['.', 'include'])
+
+security_dep = []
+if with_security
+  security_dep = dependency('libcrypto')
+endif
+
+add_project_arguments(
+  ['-D_GNU_SOURCE'],
+  language: 'c')
+
+# we will need this below
+cc = meson.get_compiler('c')
+
+#
+# build our config.h file
+#
+
+conf = configuration_data()
+conf.set('WITH_SECURITY', with_security)
+conf.set('WITH_SLP', with_slp)
+conf.set('HAVE_SLP_H', with_slp)
+conf.set('RUNDIR', run_dir)
+
+config_h = configure_file(
+  output: 'config.h',
+  configuration: conf)
+
+config_incdir = include_directories('.')
+
+config_dep = declare_dependency(
+  include_directories: config_incdir,
+  sources: config_h)
+
+#
+# build our library -- we always build either the static or
+# the shared library, or both
+#
+
+lib_deps = [config_dep, security_dep]
+if with_slp
+  lib_deps += cc.find_library('slp')
+endif
+
+if enable_static
+  libisns_lib = static_library('isns',
+    lib_srcs,
+    include_directories: incl_dirs,
+    dependencies: lib_deps,
+    install: true)
+endif
+
+if enable_shared
+  if enable_shared_version
+    shared_link_args = '-Wl,--version-script=@0@/libisns.vers'.format(meson.current_source_dir())
+  else
+    shared_link_args = []
+  endif
+  libisns_lib = shared_library('isns',
+    lib_srcs,
+    include_directories: incl_dirs,
+    dependencies: lib_deps,
+    link_args: shared_link_args,
+    version: '0',
+    install: true)
+endif
+
+libisns_dep = declare_dependency(link_with: libisns_lib)
+
+#
+# build binaries
+#
+
+our_dependencies = [config_dep, security_dep, libisns_dep]
+if with_slp
+  our_dependencies += cc.find_library('slp')
+endif
+
+executable('isnsd',
+  sources: files('isnsd.c'),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  install: true,
+  install_dir: sbin_dir)
+
+executable('isnsdd',
+  sources: files(['isnsdd.c', 'local.c']),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  install: true,
+  install_dir: sbin_dir)
+
+executable('isnsadm',
+  sources: files('isnsadm.c'),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  install: true,
+  install_dir: sbin_dir)
+
+# a testing program (not installed, not currently used here)
+e = executable('bitvector',
+  sources: files('bitvector.c'),
+  include_directories: incl_dirs,
+  dependencies: our_dependencies,
+  c_args: ['-DTEST'])
+test('test bitvector', e)
+
+# build our test files ??? (FIXME: not working)
+pyunit_test_works = false
+if pyunit_test_works
+  test_progs = []
+  foreach t: tests
+    e = executable(t,
+      sources: files('tests' / t + '.c'),
+      include_directories: incl_dirs,
+      dependencies: our_dependencies)
+    test_progs += e
+  endforeach
+endif
+
+#
+# man pages
+#
+
+foreach m: man8_pages
+  install_man(m, install_dir: man_dir / 'man8')
+endforeach
+foreach m: man5_pages
+  install_man(m, install_dir: man_dir / 'man5')
+endforeach
```

install_man will cleverly figure out the correct subdir to install each man page, based on the extension (in this case, "8"). You can even add a list of man pages for several different sections as the arguments to install_man -- it will install each argument to the right place for that specific man page.

```suggestion
install_man(man8_pages, man5_pages)
```

---

fixed